src/Propulsion/Submission.fs

```diff
@@ -59,7 +59,7 @@ type SubmissionEngine<'P, 'M, 'S, 'B when 'P : equality>
     (   log : ILogger, statsInterval,
         mapBatch : (unit -> unit) -> Batch<'P, 'M> -> struct ('S * 'B),
         submitStreams : 'S -> unit, // We continually submit the spans of events as we receive them, in order to minimise handler invocations required
-        waitToSubmitBatch : CancellationToken -> ValueTask<bool>, // We continually fill the target to the degree possible. When full, we sleep until capacity is released on completion
+        waitToSubmitBatch : CancellationToken -> Task, // We continually fill the target to the degree possible. When full, we sleep until capacity is released on completion
```

Comment:
We never used the `bool` so no need for it to be part of the signature?

Reply:
yes, removing is goof